orchestrator/careplancontributor/handle_taskfiller.go

```diff
@@ -320,30 +299,46 @@ func (s *Service) selectWorkflow(cpsClient fhirclient.Client, task *fhir.Task) (
 			if coding.System == nil || coding.Code == nil {
 				continue
 			}
-			candidateCodes = append(candidateCodes, coding)
+			var present bool
+			for _, taskReasonCode := range taskReasonCodes {
+				if taskReasonCode.System == coding.System && taskReasonCode.Code == coding.Code {
+					present = true
+					break
+				}
+			}
+			if !present {
+				taskReasonCodes = append(taskReasonCodes, coding)
+			}
 		}
 	}
-	// find matching workflow
-	var matchedCodes []string
-	var matchedWorkflow *taskengine.Workflow
-	for _, coding := range candidateCodes {
-		key := fmt.Sprintf("%s|%s", *coding.System, *coding.Code)
-		if slices.Contains(matchedCodes, key) {
-			// duplicate code, doesn't matter
+	taskReasonCodes = slices.Deduplicate(taskReasonCodes, func(a, b fhir.Coding) bool {
```

Comment:
It seems like the check above should ensure duplicates are not added to the list, if we are going to run the dedupe anyway why not add all the matching codes and skip the comparison?

Reply:
Then I'd have to write it twice (for task.reasoncode and task.reasonReference.code), and it's a bit of an elaborate Task (checking system and code aren't nil, then comparing both). So that's why it's cognitivially easier this way.

Reply:
Fine with me